File: AtCoder/KyoproTenke/cpp/018-Statue_of_Chokudai.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

ll T;
ll L, X, Y;
ll Q;
ll E;

double vt(ll E) {
  return L * 0.5 * cos(-2 * M_PI / T * E - M_PI / 2);
}

double wt(ll E) {
  return L * 0.5 * sin(2 * M_PI / T * E - M_PI / 2) + L * 0.5;
}

int main() {
  cin >> T >> L >> X >> Y >> Q;
  ll Z = 0;
  double u = 0;
  vector<double> ress;
  while (Q--) {
    cin >> E;
    double v = vt(E);
    double w = wt(E);
    double diag =
        sqrt((u - X) * (u - X) + (v - Y) * (v - Y) + (w - Z) * (w - Z));
    double res = acos(w / diag) * 180 / M_PI;
    ress.push_back(90 - res);
  }
  for (auto res : ress)
    printf("%.8f\n", res);
}

File: AtCoder/KyoproTenkei/python/018-Statue_of_Chokudai.py
from math import cos, sin, acos, degrees, pi


T = float(input())
L, X, Y = map(float, input().split())


def vt(E):
    return L * 0.5 * cos(-2 * pi / T * E - pi / 2)


def wt(E):
    return L * 0.5 * sin(2 * pi / T * E - pi / 2) + L * 0.5


Z = 0
u = 0
Q = int(input())
ress = []
for _ in range(Q):
    E = int(input())
    v = vt(E)
    w = wt(E)
    diag = ((u - X)**2 + (v - Y)**2 + (w - Z)**2)**0.5
    res = degrees(acos(w / diag))
    ress.append(90 - res)

print('\n'.join(map(lambda x: f"{x:.8f}", ress)))
